### DISCUSSION TITLE
Exception in thread "Thread-5"

### DISCUSSION THREAD

**Discussion by semmakara6666** [community/community#140374] [Programming Help]
### Cause of error
```java
Exception in thread "Thread-5" java.lang.IllegalArgumentException: Trying to execute a sync operation with a storage manager for an anonymous account
    at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:49)
    at com.owncloud.android.ui.activity.DrawerActivity.lambda$fetchExternalLinks$9(DrawerActivity.java:1231)
    at com.owncloud.android.ui.activity.DrawerActivity.$r8$lambda$oHM2j1MZkviAIdyIFWoI2kSeF9c(Unknown Source:0)
    at com.owncloud.android.ui.activity.DrawerActivity$$ExternalSyntheticLambda6.run(D8$$SyntheticClass:0)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30290290`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a02q`
* Model: `SM-A025F`
* Id: `SP1A.210812.016`
* Product: `a02qnnxx`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `A025FXXS8CXH1`

**Comment by suchakreepanyawong** [comment]
1. Check the Account Configuration
Ensure that you're logged in with a valid account. It seems like the operation is trying to use an "anonymous" account, which could indicate that your session may have expired or you haven't logged in properly.
2. Clear App Cache and Data
Go to Settings > Apps > Nextcloud > Storage, and try clearing the app’s cache and data. Afterward, log in again and see if the issue persists.
3. Reinstall the App
If clearing cache and data doesn't resolve the issue, try uninstalling and reinstalling the app. This will ensure you have a clean installation and might fix any corrupted files or configurations.
4. Check for App and System Updates
Ensure that the Nextcloud app and your device's firmware are up to date. Sometimes bugs are fixed in newer versions of the app, or a system update may resolve compatibility issues.
5. Inspect the Code (if you’re a developer)
If you’re developing or debugging the app:

SyncOperation.java:49: Check what happens when the sync operation is executed and ensure that a valid account is being passed to the storage manager.
DrawerActivity.java:1231: Investigate how the fetchExternalLinks method handles the account and why an anonymous account might be used.
6. Report the Issue
If this is a persistent bug, you might want to report it to the Nextcloud GitHub repository or their support team. Include the stack trace and app/device information for better assistance.

**Reply by ebndev** [reply]
Hi @suchakreepanyawong, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @semmakara6666 ! We're happy you're here. You are more likely to get a useful response if you have a clear title and are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by KavinMadusanka** [comment]
Step 1: Check Account Settings
Open the app on your device.
Ensure that you are logged into an active account and not using an anonymous or guest session.
* Navigate to Settings in the app.
* Check if the account credentials are set properly.
* If the app shows that you're logged in as an anonymous user or a guest, log out and log back in with your proper credentials.

Step 2: Clear Cache and Data (Optional)
If logging in again doesn’t resolve the issue, try clearing the app's cache and data:

* Go to Settings on your device.
* Select Apps.
* Find and select the Nextcloud app (com.nextcloud.client).
* Tap Storage.
* Select Clear Cache and then Clear Data.
* Relaunch the app and log in again with your credentials.

Step 3: Check App Permissions
Ensure that the app has proper permissions to access storage:

* Go to Settings > Apps > Nextcloud.
* Tap Permissions.
* Ensure that Storage permission is granted.
* If not, enable the necessary permissions.

Step 4: Reinstall the App
If the issue persists:

* Uninstall the app by going to Settings > Apps > Nextcloud > Uninstall.
* Download the app again from the Google Play Store and reinstall it.
* Log in to the app with your credentials and check if the sync operation works correctly.

Step 5: Check for App Updates
Ensure that you're using the latest version of the Nextcloud app:

* Go to the Google Play Store.
* Search for Nextcloud.
*If there is an update available, update the app and check if the issue is resolved.

Step 6: Look for Server-Specific Settings
If this issue is related to your server configuration (self-hosted), double-check if the server supports sync for your account. In some cases, account misconfigurations on the server side could lead to this issue.

**Reply by ebndev** [reply]
Hi @KavinMadusanka, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟
